code-challenges.test.js

```diff
@@ -12,35 +14,117 @@
 
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
-// a) Create a test with an expect statement using the variable provided.
+// Input: create a test with the given variables
+// Output: create a function that will allow the test to pass
+// First I'm going to create a jest test using the given array "people" and create a describe function "newArray" with it and expect statements.
+// Next create a function called newArray that takes in an array of objects as its param
+//use map to iterate over each object in the array
+//create a new variable called "name" that replaces the first letter of each name with the capatalized version.
+//return a sentence for each person using string interpolation
+//
+
 
+// a) Create a test with an expect statement using the variable provided.
+describe("newArray", () => {
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
+  { name: "arthur dent", occupation: "a radio employee" } ]
+
+it("Takes an array of objects and return an array with a sentence about each person with their name capatalized", () => {
+expect(newArray(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+})
+})
+
+//Output: ReferenceError: newArray is not defined
+
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 // b) Create the function that makes the test pass.
 
+const newArray = (people) => {
+  return people.map((person) => {
+    const name = person.name.replace(/\b\w/g, (word) => word.toUpperCase());
+    // in the variable name I am using "person as the object" and the .replace method to remove the original letter and replace it with the capatalized version. bwg is matching and word characters that occur at the beginning of a word bondary (name) and g matches it globally. 
```

Comment:
Great job adding this pseudocode here which helps explain how you are handling the object or "person". 👏🏽